Assets/MixedRealityToolkit/Definitions/MixedRealityInputDataProviderConfiguration.cs

```diff
@@ -39,12 +39,14 @@ public struct MixedRealityInputDataProviderConfiguration : IMixedRealityServiceC
         [SerializeField]
         private BaseMixedRealityProfile deviceManagerProfile;
 
+        /// <inheritdoc />
+        public BaseMixedRealityProfile Profile => deviceManagerProfile;
+
         /// <summary>
         /// Device manager specific configuration profile.
         /// </summary>
         public BaseMixedRealityProfile DeviceManagerProfile => deviceManagerProfile;
```

Comment:
Should this be marked `Obsolete`, like you did in MixedRealityServiceConfiguration?

Reply:
>No because they all cast to the appropriate profile type. InputSystem is just weird that it does not have a unique BaseInputConfiguration class. ServiceConfiguration is meant to be generic and hence why I obsolete that one

Ohh I see now that the others have specific types, but this one is still `BaseMixedRealityProfile`.

Reply:
I think this is my only remaining question, where it looks like both properties here use the same profile type

Reply:
oh I was wondering why this kept getting unresolved lol

I don't want to change it in InputConfiguration because really there is a separate bug in that there should be an InputDataConfiguration type. InputConfiguration should follow the model of the other dataprovider owners vs removing the property here. 

I could add that in this PR....I am just trying to restraint myself from adding more and more fixes/code in PRs to keep them small lol

While it doesn't make sense to make one for the service since it's supposed to be generic for any extension service

I'll open a bug to track it

Reply:
Bug filed: #6900